rcl_action/src/rcl_action/names.c

```diff
@@ -0,0 +1,112 @@
+// Copyright 2018 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#include "rcl_action/names.h"
+
+#include "rcl/error_handling.h"
+#include "rcutils/format_string.h"
+
+rcl_ret_t
+rcl_action_get_goal_service_name(
+  const char * action_name,
+  rcl_allocator_t allocator,
+  char ** goal_service_name)
+{
+  RCL_CHECK_ALLOCATOR_WITH_MSG(&allocator, "allocator is invalid", return RCL_RET_INVALID_ARGUMENT);
+  RCL_CHECK_ARGUMENT_FOR_NULL(action_name, RCL_RET_INVALID_ARGUMENT, allocator);
+  RCL_CHECK_ARGUMENT_FOR_NULL(goal_service_name, RCL_RET_INVALID_ARGUMENT, allocator);
+  *goal_service_name = rcutils_format_string(allocator, "%s/_action/send_goal", action_name);
+  if (*goal_service_name == NULL) {
```

Comment:
style nit: prefer variable on rhs to avoid accidental assignment (ie `if (NULL == *goal_service_name)`). [Ref](https://index.ros.org/doc/ros2/Developer-Guide/#c).

Same for other equality checks below.

Reply:
Good catch. Fixed.